packages/blockchain-wallet-v4-frontend/src/hooks/useFetcher/useFetcher.tsx

```diff
@@ -0,0 +1,50 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import { FetcherHookData } from './useFetcher.types'
+
+const useFetcher = <T extends unknown = unknown, E extends object = object>(path: string) => {
+  const [state, setState] = useState<FetcherHookData<T, E>>({
+    data: undefined,
+    isFetching: false,
+    isLoading: false
+  })
+
+  const startFetch = useCallback(
+    async (path: string) => {
+      try {
+        setState((state) => ({
+          data: state.data,
+          error: undefined,
+          isFetching: true,
+          isLoading: state.data === undefined
+        }))
+
+        const response = await fetch(path)
+
+        const data: T = await response.json()
+
+        setState({
+          data,
+          isFetching: false,
+          isLoading: false
+        })
+      } catch (error) {
+        setState({
+          data: undefined,
+          error,
+          isFetching: false,
+          isLoading: false
+        })
+      }
+    },
+    [setState]
+  )
+
+  useEffect(() => {
+    startFetch(path)
+  }, [path])
+
+  return state
+}
+
+export default useFetcher
```

Comment:
if we think returning a `Remote` type is overkill, perhaps we also put that in the name of the hook.  maybe `useNonRemoteFetch` or something?

Reply:
if we split into two maybe the names should be

`useFetchNonRemote`
`useFetchRemote`

Reply:
I used the useFetcher name because it is just a wrapper for the fetch function.
I think it should not have the Remote name because it doesn't know anything about Remote type and if one day we remove the Remote dependency from the project we would need to change the name of this hook too, right?